music21/expressions.py

```diff
@@ -517,21 +519,205 @@ def fillListOfRealizedNotes(
+        >>> invertedMordent.name
+        'inverted mordent (sharp)'
+
+        '''
+        theName: str = super().name
+        if self.accid is not None:
+            theName += ' (' + self.accid + ')'
+        return theName
+
+    @property
+    def accid(self) -> str | None:
+        return self._accid
+
+    @accid.setter
+    def accid(self, newAccid: str | None):
+        if self._size is not None:
+            raise ExpressionException('Cannot set mordent accid if size already set.')
+
+        # silently ignore invalid accid
+        if newAccid is None:
+            self._accid = None
+        elif pitch.isValidAccidentalName(newAccid):
+            self._accid = pitch.makeAccidentalName(newAccid)
+
+    @property
+    def size(self) -> interval.IntervalBase | None:
+        if self._size is None:
+            raise ExpressionException(
+                'Cannot compute mordent size without srcPitch unless .size has been '
+                + 'explicitly set (use mordent.getSize(srcPitch) instead.)'
+            )
+        return self._size
+
+    @size.setter
+    def size(self, newSize: interval.IntervalBase | None):
+        if self._accid is not None:
+            raise ExpressionException('Cannot set mordent size if accid already set.')
+        self._size = newSize
+
+    @property
+    def direction(self) -> str:
+        return self._direction
+
+    def getSize(
+        self,
+        srcPitch: pitch.Pitch,
+    ) -> interval.IntervalBase:
+        if self._size is not None:
+            return self._size
+
+        if self.direction not in ('up', 'down'):
+            raise ExpressionException('Cannot compute mordent size if I do not know its direction')
+
+        otherPitch: pitch.Pitch = copy.deepcopy(srcPitch)
+        otherPitch.accidental = None
+
+        if self.direction == 'up':
+            otherPitch.transpose(interval.GenericInterval(2), inPlace=True)
+        else:
+            otherPitch.transpose(interval.GenericInterval(-2), inPlace=True)
+
+        if self.accid is not None:
+            otherPitch.accidental = pitch.Accidental(self.accid)
+        return interval.Interval(srcPitch, otherPitch)
+
+    def resolveOtherPitches(self, srcPitch: pitch.Pitch):
+        if self._otherPitches:
+            # otherPitches have already been resolved
+            return
+
+        transposeInterval: interval.IntervalBase = self.getSize(srcPitch)
+        otherPitch: pitch.Pitch = srcPitch.transpose(transposeInterval, inPlace=False)
+        if self.accid is not None:
+            if otherPitch.accidental is None:
+                otherPitch.accidental = pitch.Accidental(0)
+            otherPitch.accidental.displayStatus = True
+        self._otherPitches = (otherPitch,)
+
+    @property
+    def otherPitch(self) -> pitch.Pitch | None:
+        if self._otherPitches:
+            return self._otherPitches[0]
+
+    def realize(self, srcObj: note.Note, *, inPlace=False):
```

Comment:
Q -- can ornaments appear on Chords/Rests/Unpitched?  Should this be checked for?  (I think it's an overall music21 bug, so maybe not.

Reply:
a trill on unpitched, yes.  It'd be the one trill to have a size of unison.  It's essentially a timpani roll.

Reply:
I think I have to make another change to get an Unpitched Trill (e.g. snare drum roll) to work properly.  I don't know how a timpani roll would work in music21.  It's pitched, so a Trill would imply changing pitch.  Which ain't what a timpani roll is supposed to do.

Reply:
Done.

Reply:
I wonder if eventually "Roll" should be a separate class from Trill -- but can have "displayAsTrill" set or something.  They use the same notation, but conceptually so different.